qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/pairing_table_widget/pairing_table_widget_presenter.py

```diff
@@ -220,28 +262,173 @@ def remove_selected_rows_in_view_and_model(self, pair_names):
+            group_selector_widget.currentIndexChanged.connect(lambda: self._view.on_cell_changed(row, 2))
+        elif group == "group_2":
+            group_selector_widget.currentIndexChanged.connect(lambda: self._view.on_cell_changed(row, 3))
+
+        self._view.set_combo_box_index(group_selector_widget, entry)
+        self._view.set_widget_in_table(row, col, group_selector_widget)
+
+    def _add_alpha_entry(self, row, col, entry):
+        """Add an alpha entry to the table."""
+        alpha_widget = table_utils.ValidatedTableItem(self._view.validate_alpha)
+        alpha_widget.setText(entry)
+        self._view.set_item_in_table(row, col, alpha_widget)
+
+    def _add_to_analyse_checkbox(self, item, entry):
+        """Set the 'to_analyse' checkbox state."""
+        self._view.set_checkbox_in_table(item, checked=bool(entry))
+
+    def _add_guess_alpha_button(self, row):
+        """Add the 'Guess Alpha' button to the last column."""
+        guess_alpha_widget = self._view.guess_alpha_button
+        guess_alpha_widget.clicked.connect(self._view.guess_alpha_clicked_from_row)
+        self._view.set_widget_in_table(row, 5, guess_alpha_widget)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Context Menu
+    # ------------------------------------------------------------------------------------------------------------------
+    def _context_menu_event(self, _event):
+        """Overridden method for dealing with the right-click context menu"""
+        menu = self._view.create_context_menu
+
+        add_pair_action = self._context_menu_add_pair_action(self._view.add_pair_button.clicked.emit)
+        remove_pair_action = self._context_menu_remove_pair_action(self._view.remove_pair_button.clicked.emit)
+
+        if self._view.is_disabled:
+            self._view.add_pair_action.setEnabled(False)
+            self._view.remove_pair_action.setEnabled(False)
+        # set-up the menu
+        menu.addAction(add_pair_action)
+        menu.addAction(remove_pair_action)
+        menu.popup(self._view.cursor_position)
+
+    def _context_menu_add_pair_action(self, slot):
+        add_pair_action = self._view.get_add_pair_action
+        add_pair_action.setCheckable(False)
+        if len(self.get_selected_row_indices()) > 0:
+            add_pair_action.setEnabled(False)
+        add_pair_action.triggered.connect(slot)
+        return add_pair_action
+
+    def _context_menu_remove_pair_action(self, slot):
+        if len(self.get_selected_row_indices()) > 1:
+            # use plural if >1 item selected
+            remove_pair_action = self._view.get_remove_pair_action(True)
+        else:
+            remove_pair_action = self._view.get_remove_pair_action()
+        if self._view.get_pairing_table.rowCount() == 0:
+            remove_pair_action.setEnabled(False)
+        remove_pair_action.triggered.connect(slot)
+        return remove_pair_action
+
```

Comment:
These functions, even though they contain a bit of logic, they are creating q elements and connecting slots/signals, perhaps is better to move them to the view.